tools/config_explorer.cxx

```diff
@@ -34,81 +34,61 @@
-            << "\n"
-            << "Options are:\n"
-            << "  -h / --help      displays usage information\n"
-            << "  --path name      add directory to config search path(can appear multiple times)\n"
-            << "  -Iname           add directory to config search path(can appear multiple times)\n"
-            << "  -ds              generate detailed application-specific search paths\n"
-            << "  -dc              generate detailed config contents output\n"
-            << "  -a name          alternate application name\n"
-            << "  -v version       optional application version string\n"
-            << "  --prefix dir     optional non-standard install prefix directory\n"
-            << "\n"
-            << "If -ds is specified, the detailed search paths that apply to the application are\n"
-            << "displayed only otherwise, the config file is loaded.\n"
-            << "\n"
-            << "The option -dc only has effect when a config file is specified and causes a\n"
-            << "detailed output of the config entries.\n"
-            << "\n"
-            << "If -I or --path are specified, then the config file is only searched for using\n"
-            << "the specified path. The application name based paths are not used.\n"
-  ;
-
-  return;
-}
-
-
-// ------------------------------------------------------------------
-static int
-path_callback( const char*  argument,   // name of argument
-               const char*  value,      // value of argument
-               void*        call_data ) // data from register call
+add_command_options()
 {
-  const std::string p( value );
+  m_cmd_options->custom_help( wrap_text( "[options] config-file\n"
+    "This program assists in debugging config loading problems. It loads a "
+    "configuration and displays the contents or displays the search path."
+    "Additional paths can be specified in \"KWIVER_CONFIG_PATH\" environment variable"
+    "or on the command line with the -I or --path options."
+    "If -ds is specified, the detailed search paths that apply to the application are"
+    "displayed only, otherwise the config file is loaded."
+    "\n\n"
+    "The option -dc only has effect when a config file is specified and causes a"
+    "detailed output of the config entries."
+    "\n\n"
+    "If -I or --path are specified, then the config file is only searched for using"
+    "the specified path. The application name based paths are not used."
+                                ) );
+
+ m_cmd_options->positional_help( "\n  config-file  - name of configuration file." );
+
+  m_cmd_options->add_options()
+    ( "h,help", "Display applet usage" )
+    ( "ds", "Display detailed application search path" )
+    ( "dc", "Display detailed config contents" )
+    ( "I,path", "Add directory to config search path", cxxopts::value<std::vector<std::string>>() )
+    ( "a,application", "Application name", cxxopts::value<std::string>() )
```

Comment:
Maybe this is outside the scope of this branch, but should `application` default to "kwiver" and `version` default to the version string in `version.h`?

I'm planning to port over some default config files from MAP-Tk soon.  These will get installed to something like `/usr/share/kwiver/[version]/config/` and the `read_config_file` variant that takes these values is designed to add that to the config search path.  This provides a place to store and look for default configuration for applets and algorithms without having to hard code it.

Reply:
I think that the default app and version should be with the actual applet rather than with the config explorer. Also the applet name could be used as the application name. Could add a method to the base applet to ``read_applet_config( ... ) `` that could supply the app name and version string.